react-validated-input-component/src/validate-input.jsx

```diff
@@ -0,0 +1,72 @@
+import React, { useState } from 'react';
+
+export default function ValidateInput() {
+
+  const switchClasses = {
+    hidden: 'hidden',
+    greenCheck: {
+      style: 'fa-sharp fa-solid fa-check green'
+    },
+    redX: {
+      style: 'fa-sharp fa-solid fa-xmark red',
+      text: 'Your password is too short.'
+    }
+  };
+
+  const [password, setPassword] = useState('');
+
+  function handleSubmit(event) {
+    event.preventDefault();
+  }
+
+  const conditions = ['!', '@', '#', '$', '%', '^', '&', '*', '(', ')'];
+  const zeroToNine = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+  const special = conditions.some((char) => password.includes(char));
+  const numbers = zeroToNine.some((number) => password.includes(number));
+
+  if (password.length < 8 && !special && !numbers) {
+    switchClasses.redX.text = 'Password requires a number and special chararcteer.';
+  } else if (password.length >= 8 && !special && !numbers) {
+    switchClasses.redX.text = 'Password requires a number and special chararcteer.';
+  } else if (password.length < 8 && !special) {
+    switchClasses.redX.text = 'Password requires a special chararcteer.';
+  } else if (password.length >= 8 && !special) {
+    switchClasses.redX.text = 'Password requires a special chararcteer.';
+  } else if (password.length < 8 && !numbers) {
+    switchClasses.redX.text = 'Password requires a number.';
+  } else if (password.length >= 8 && !numbers) {
+    switchClasses.redX.text = 'Password requires a number.';
+  } else if (password.length < 8) {
+    switchClasses.redX.text = 'Your password is too short.';
+  }
+
+  if (password.length === 0) {
+    switchClasses.hidden = 'hidden';
+  } else {
+    switchClasses.hidden = '';
+  }
+
+  const classes = switchClasses[password.length >= 8 && special && numbers ? 'greenCheck' : 'redX'];
```

Comment:
I like this. I would suggest, though, that you destructure the result to make the code below more readable.

For example, `const { style, text } = ...` and `className={style}` are more readable.